cli/generate.go

```diff
@@ -477,3 +490,20 @@ func loadJWTSecret(from string) (absFrom string, err error) {
 	}
 	return
 }
+
+func validateLido(network string, flags *GenCmdFlags) error {
+	if !flags.noMev {
+		_, ok := mevboostrelaylist.DeployedContractAddresses[network]
+		if !ok {
+			options := mevboostrelaylist.GetLidoSupportedNetworksMevBoost()
+			return fmt.Errorf("invalid network: Choose valid network for Lido with MEV-Boost: %v", options)
+		}
+	}
+	_, ok := contracts.FeeRecipient[network]
+	if !ok {
+		options := contracts.GetLidoSupportedNetworks()
+		return fmt.Errorf("invalid network: Choose valid network for Lido: %v", options)
+	}
+
+	return nil
+}
```

Comment:
On the other hand, every `nodeSetup` would have some validation logic that gets the `network`, the settings (`flags`), and return an error. This is very interface-able.

Reply:
[Done](https://github.com/NethermindEth/sedge/pull/389/commits/ca7ad5e978436664ee6f1f626f4b03febff5a263)